File: backend/java/Vehicle_Configurator/src/main/java/com/example/demo/controllers/InvoiceController.java
package com.example.demo.controllers;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import com.example.demo.dto.InvoiceRequestDTO;
import com.example.demo.dto.InvoiceResponseDTO;
import com.example.demo.entities.Invoice;
import com.example.demo.services.InvoiceService;

@RestController
@RequestMapping("/api/invoice")
@CrossOrigin(origins = "http://localhost:5173")
public class InvoiceController {

    @Autowired
    private InvoiceService invoiceService;


    @PostMapping("/generate")
    public Invoice generateInvoice(@RequestBody InvoiceRequestDTO request) {

        return invoiceService.saveInvoice(request);
    }
    @GetMapping("/{invoiceId}")
    public InvoiceResponseDTO getInvoice(
            @PathVariable int invoiceId){

        return invoiceService.getInvoice(invoiceId);

    }
}

File: backend-dotnet/Controllers/InvoiceController.cs

using backend_dotnet.DTO;
using backend_dotnet.Models;
using backend_dotnet.Services;
using Microsoft.AspNetCore.Mvc;

namespace backend_dotnet.Controllers
{
    [ApiController]
    [Route("api/invoice")]
    public class InvoiceController : ControllerBase
    {
        private readonly IInvoiceService invoiceService;

        public InvoiceController(IInvoiceService invoiceService)
        {
            this.invoiceService = invoiceService;
        }

        [HttpPost("generate")]
        public ActionResult GenerateInvoice(
    [FromBody] InvoiceRequestDTO request)
        {
            Invoice invoice = invoiceService.SaveInvoice(request);

            return Ok(new
            {
                invoiceId = invoice.InvoiceId
            });
        }

        [HttpGet("{invoiceId:int}")]
        public ActionResult<InvoiceResponseDTO> GetInvoice(
            int invoiceId)
        {
            return invoiceService.GetInvoice(invoiceId);
        }
    }
}

